browser/brave_news/brave_news_tab_helper.cc

```diff
@@ -144,9 +218,17 @@ void BraveNewsTabHelper::RemoveObserver(PageFeedsObserver* observer) {
   observers_.RemoveObserver(observer);
 }
 
+bool BraveNewsTabHelper::ShouldFindFeeds() {
+  auto* prefs = Profile::FromBrowserContext(web_contents()->GetBrowserContext())
+                    ->GetPrefs();
+  return brave_news::GetIsEnabled(prefs) &&
+         prefs->GetBoolean(brave_news::prefs::kShouldShowToolbarButton);
+}
+
 void BraveNewsTabHelper::AvailableFeedsChanged() {
-  for (auto& observer : observers_)
-    observer.OnAvailableFeedsChanged(GetAvailableFeeds());
+  for (auto& observer : observers_) {
+    observer.OnAvailableFeedsChanged(GetAvailableFeedUrls());
+  }
```

Comment:
I think moving `GetAvailableFeedUrls()` out of the loop would be better so that we can avoid redundant feeds building. Or we might need to caching the feed urls on navigation happens (doesn't need to be in this PR) 
```suggestion
  const auto& feeds = GetAvailableFeedUrls();
  for (auto& observer : observers_) {
    observer.OnAvailableFeedsChanged(feeds));
  }
```

Reply:
Great call!